### PR TITLE
Login and redirect

### PR DESCRIPTION
**Pull request by tspr0tege**
Routes added to controllers
Signup creates a new user and has it's own dedicated route through the api.
Login will not proceed without a matching username and password in DB.
Successful login redirects to /bands

### CODE REVIEW

File: client/src/modals/login.jsx [hunk 1]
```diff
@@ -1,25 +1,44 @@
...
+          expires: expires.toGMTString(),
+        });
+        history.push('/bands');
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+    // email.value = '';
+    // password.value = '';
     cb();
```
Comment (zkodjo) [inline client/src/modals/login.jsx]: What does this cb() do?

File: client/src/modals/login.jsx [hunk 2]
```diff
@@ -1,25 +1,44 @@
...
+        const expires = new Date();
+        expires.setMinutes(expires.getMinutes() + 1);
+        cookies.set({
+          userId: response.data.id,
+          expires: expires.toGMTString(),
+        });
+        history.push('/bands');
+      })
+      .catch((error) => {
+        console.log(error);
```
Comment (zkodjo) [inline client/src/modals/login.jsx]: Can you return this as console.error instead of console.log?

File: client/src/modals/signup.jsx [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+      password: password.value,
+      description: '',
+      genre: genres[0],
+    };
+    axios.post('/login', data)
+      .then((response) => {
+        console.log(response);
+      })
+      .catch((error) => {
+        console.log(error);
```
Comment (zkodjo) [inline client/src/modals/signup.jsx]: change to console.error

File: client/src/modals/signup.jsx [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    const genres = [];
+    for (let i = 0; i < checkboxes.length; i += 1) {
+      genres.push(checkboxes[i].value);
+    }
+    const data = {
+      name: bandname.value,
+      email: email.value,
+      password: password.value,
+      description: '',
+      genre: genres[0],
```
Comment (zkodjo) [inline client/src/modals/signup.jsx:27]: Does this only send the first genre in the array? Should we redesign how the signup endpoint works to handle this?

**Review by zkodjo — CHANGES_REQUESTED** [review]
I put some comments inline. Do you want me to redesign the signup endpoint somehow?

File: client/src/modals/login.jsx (hunk 1 above)
Comment (tspr0tege) [reply client/src/modals/login.jsx]: Honestly do not know. I don't remember adding it there, so it's getting snapped out of existence.

**Review by zkodjo — APPROVED** [review]
Approved, thanks!

**Pull request merged**